packages/nvidia_nat_core/tests/nat/middleware/test_timeout_middleware.py

```
@@ -0,0 +1,234 @@
+# ==================== Single Invocation Tests ====================
+
+
+class TestTimeoutMiddlewareInvoke:
+    """Tests for function_middleware_invoke timeout enforcement."""
+
+    @pytest.mark.asyncio
+    async def test_completes_within_timeout(self, mock_builder, function_context):
+        """Function that completes within the timeout returns normally."""
+        middleware: TimeoutMiddleware = _make_middleware(mock_builder, timeout=5.0)
+
+        async def fast_function(*args, **kwargs):
```

Ruff (0.15.2)

[warning] Unused function argument: `args`

(ARG001)